extensions/mssql/src/credentialstore/credentialstore.ts

```diff
@@ -37,7 +40,87 @@ class CredentialsFeature extends SqlOpsFeature<any> {
 		});
 	}
 
-	protected registerProvider(options: any): Disposable {
+	/**
+	 * Removes a credential for a given connection profile from sqltoolsservice
+	 * and adds it to keytar
+	 */
+	private async cleanCredential(conn: azdata.connection.ConnectionProfile): Promise<boolean> {
+		const credentialId = Utils.formatCredentialId(conn);
+		// read password from every saved password connection
+		const credential = await this._client.sendRequest(Contracts.ReadCredentialRequest.type, { credentialId, password: undefined });
+		if (credential.password) {
+			const password = credential.password;
+			// save it in keychain
+			await this.keychain.setPassword(credentialId, password);
+			// check if it's saved
+			const savedPassword = await this.keychain.getPassword(credentialId);
+			if (savedPassword === password) {
+				// delete from tools service
+				const result = await this._client.sendRequest(Contracts.DeleteCredentialRequest.type,
+					{ credentialId, password: undefined });
+				return result;
+			} else {
+				return false;
+			}
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Try to migrate all the credentials to the keytar system
+	 * @returns
+	 */
+	private migrateCredentials(): Thenable<boolean> {
+		return azdata.connection.getConnections(false).then((connections) => {
+			const savedPasswordConnections = connections.filter(conn => conn.savePassword === true);
+			return new Promise(async (resolve, reject) => {
+				for (let i = 0; i < savedPasswordConnections.length; i++) {
+					let conn = savedPasswordConnections[i];
+					await this.cleanCredential(conn);
+				}
+				await Utils.removeCredentialFile();
+				resolve(true);
+			});
+		});
+	}
+
+	private registerKeytarProvider(options: any): Disposable {
```

Comment:
options doesn't appear to be used here - remove

And actually - I don't see the options being used anywhere so unless I'm missing something you should just remove them all.